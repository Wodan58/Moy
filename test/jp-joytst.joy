%INCLUDE stdlib.joy

DEFINE
joy0s == newline "joy0s :" putchars newline [ [ [ joy0s body joy0s ] [ [ ] ] [
pop pop pop ] [ cons pop cons ] [ opcase pop opcase ] [ body pop body ] [ i pop
joy0s ] [ step pop [ joy0 ] cons step ] [ [ ] cons i ] ] opcase i ] step ;
joy0l == newline "joy0l :" putchars newline [ dup put [ [ joy0l body joy0l ] [
[ ] ] [ pop pop pop ] [ cons pop cons ] [ opcase pop opcase ] [ body pop body ]
[ i pop joy0l ] [ step pop [ joy0 ] cons step ] [ [ ] cons i ] ] opcase i ]
step ;
joy0 == [ [ [ joy0 body joy0 ] [ [ ] ] [ pop pop pop ] [ cons pop cons ] [
opcase pop opcase ] [ body pop body ] [ i pop joy0 ] [ step pop [ joy0 ] cons
step ] [ [ ] cons i ] ] opcase i ] step ;
END

0 setautoput .
2 3 + "\nfinal answer = " putchars put_int nl .
[ 2 3 + ] joy0s "\nfinal answer = " putchars put_int nl .
[ [ 2 3 + ] joy0s ] joy0s "\nfinal answer = " putchars put_int nl .
[ [ [ 2 3 + ] joy0s ] joy0s ] joy0s "\nfinal answer = " putchars put_int nl .
[ 2 3 + ] joy0l "\nfinal answer = " putchars put_int nl .
[ [ 2 3 + ] joy0l ] joy0l "\nfinal answer = " putchars put_int nl .
[ [ [ 2 3 + ] joy0l ] joy0l ] joy0l "\nfinal answer = " putchars put_int nl .
[ [ [ [ [ 2 3 + ] joy0 ] joy0 ] joy0 ] joy0 ] joy0 "\nfinal answer = " putchars
put_int nl .
