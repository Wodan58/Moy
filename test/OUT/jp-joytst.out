
final answer = 5 

joy0s :

final answer = 5 

joy0s :

joy0s :

joy0s :

joy0s :
joy0s :


final answer = 5 

joy0s :

joy0s :

joy0s :

joy0s :
joy0s :


joy0s :

joy0s :

joy0s :
joy0s :


final answer = 5 

joy0l :
2 3 + 
final answer = 5 

joy0l :
[2 3 +] joy0l 
joy0l :
newline 
joy0l :
'\n putch 
"joy0l :" putchars joy0l :newline 
joy0l :
'\n putch 
[dup put [[joy0l body joy0l] [[]] [pop pop pop] [cons pop cons] [opcase pop opcase] [body pop body] [i pop joy0l] [step pop [joy0] cons step] [[] cons i]] opcase i] step 2 3 + 
final answer = 5 

joy0l :
[[2 3 +] joy0l] joy0l 
joy0l :
newline 
joy0l :
'\n putch 
"joy0l :" putchars joy0l :newline 
joy0l :
'\n putch 
[dup put [[joy0l body joy0l] [[]] [pop pop pop] [cons pop cons] [opcase pop opcase] [body pop body] [i pop joy0l] [step pop [joy0] cons step] [[] cons i]] opcase i] step [2 3 +] joy0l 
joy0l :
newline 
joy0l :
'\n putch 
"joy0l :" putchars joy0l :newline 
joy0l :
'\n putch 
[dup put [[joy0l body joy0l] [[]] [pop pop pop] [cons pop cons] [opcase pop opcase] [body pop body] [i pop joy0l] [step pop [joy0] cons step] [[] cons i]] opcase i] step 2 3 + 
final answer = 5 

final answer = 5 
